src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -0,0 +1,123 @@
+from django.contrib.gis.geos import MultiPolygon, Polygon
+from django.http import QueryDict
+from typing import TypedDict
+
+import json
+
+
+# A list of coordinates representing a polygon.
+class PolygonFromUrlParams(TypedDict):
+    coordinates: list[tuple[float, float]]
+
+
+# A project area composed of multiple disjoint polygons.
+class ProjectAreaFromUrlParams(TypedDict):
+    # Project ID
+    id: int
+    # SRID
+    srid: int
+    # Disjoint polygons that are part of the project area.
+    polygons: list[PolygonFromUrlParams]
+
+
+# Gathers forsys input parameters from url params, database lookups, or a
+# combination of the two.
+# Of note, the option to set all forsys input paramters via url parameters is
+# intended for backend debugging purposes while the option to set most forsys
+# input parameters via database lookups is intended for production.
+class ForsysScenarioSetRequestParams():
+    # Constants for parsing url parameters.
+    _URL_USE_ONLY_URL_PARAMS = 'set_all_params_via_url_with_default_values'
+    _URL_REGION = 'region'
+    _URL_PRIORITIES = 'priorities'
+    _URL_PROJECT_AREAS = 'project_areas'
+
+    # Constants that act as default values when parsing url parameters.
+    _DEFAULT_REGION = 'sierra_cascade_inyo'
+    _DEFAULT_PRIORITIES = ['fire_dynamics',
+                           'forest_resilience', 'species_diversity']
+
+    # TODO: make regions and priorities enums to make error checking easier.
+    # TODO: add fields for constraints, costs, treatments, and thresholds.
+    # If true, additional debug information is sent to the HTTP response.
+    save_debug_info: bool
+    # The planning region.
+    region: str
+    # Conditions whose AP scores will be considered when ranking projects.
+    priorities: list[str]
+    # Project areas to be ranked. A project area may be a multipolygon.
+    project_areas: dict[int, MultiPolygon]
```

Comment:
is this keyed by the ProjectArea id? if so, can we note that here

Reply:
updated.